src/components/Dashboard.tsx

```diff
@@ -10,32 +13,84 @@ import { MobileView } from "./FullScreenViews";
 import { IS_MOBILE } from "./responsive";
 import { useLogout } from "../NostrAuthContext";
 import { LoadNode } from "../dataQuery";
+import { execute } from "../executor";
+import { planUpdateWorkspaces, usePlanner } from "../planner";
+import { KIND_WORKSPACES } from "../nostr";
+import { useApis } from "../Apis";
+
+type StorePreLoginData = (eventKinds: List<number>) => void;
+
+const StorePreLoginDataContext = React.createContext<
```

Comment:
feels like this lines of code would fit better in a separate .tsx, like LoadNode